andela/mrm_api
# #163750925 Allow duplicate room names if in different locations

**What does this PR do?**
- It allows the creation of rooms with duplicate names if they are in different locations

**Description of the task to be completed?**
- An admin should be able to create a room even if the supplied name already exists but in a different location

**How should this be manually tested?**
- make sure you run migration
- use the sample mutation below to create a room:  `ensure the locationId exists in the db`
```
mutation {
  createRoom(
            name: "Akwaba", roomType: "Meeting", capacity: 4, roomTags: [1], locationId: 1,
            calendarId:"andela.com_3630363835303531343031@resource.calendar.google.com",
            imageUrl: "https://www.officelovin.com/wp-content/uploads/2016/10/andela-office-main-1.jpg") { 
            room {
                id
                name
                roomType
                capacity
                locationId
                imageUrl
                roomTags {
                  name
                  description
                }
            }
        }
}

```
- after successfully creating the room, try creating the room again but change the calendarId to another valid one to avoid: `CalendarId already exists` error
- you should receive an error: `<room name> already exists`
- now, manually change the locationId of the room you initially created in the DB to a different one and try running the `createRoom` mutation as it is again
- you should receive a success message

**Any background context you want to provide?**
  Currently, a room created with the name `Ikogosi` in Lagos cannot be created with the same name (`Ikogosi`) in Kampala. We want our room names to be unique only based on the location.

**What are the relevant pivotal tracker stories?**

[#163750925](https://www.pivotaltracker.com/n/projects/2154921/stories/163750925)

Checklist:
